save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/controller/HeartbeatController.kt

```diff
@@ -126,3 +214,18 @@ class HeartbeatController(private val agentService: AgentService,
             .subscribe()
     }
 }
+
+/**
+ * Background inspector, which detect crashed agents
+ *
+ * @property heartbeatController
+ */
+class HeartBeatInspector(private val heartbeatController: HeartbeatController, private val heartBeatInspectorInterval: Long) : Thread() {
+    override fun run() {
+        while (!interrupted()) {
+            heartbeatController.determineCrashedAgents()
+            heartbeatController.processCrashedAgents()
+            heartbeatController.saveFromInterruptionSleep(heartBeatInspectorInterval)
```

Comment:
You can use spring mechanisms for scheduled task execution with `@Scheduled` annotation instead of manual time management

Reply:
But in this case, value should be compiler time known, so couldn't be configured from `.properties`, as I see

We use different values in production code and in tests, also it might be not hard-coded in future too

Reply:
You can use SpEL to access properties in annotation parameters, like https://www.baeldung.com/spring-scheduled-tasks#parameterizing-the-schedule

Reply:
Yeah, thanks, added